src/Workspaces/Core/Portable/Workspace/Solution/Document.cs

```diff
@@ -200,6 +202,8 @@ public Task<SyntaxTree> GetSyntaxTreeAsync(CancellationToken cancellationToken =
 
         /// <summary>
         /// Gets the root node of the current syntax tree if it is available.
+        /// In almost all cases, you should call <see cref="GetSyntaxRootAsync"/> to fetch the root node, which will parse
```

Comment:
This method was added for a very special purpose. In hindsight, we either should have named it to something more scary-sounding or just kept it internal. None of our samples will use this -- it's just a matter of people browsing the API in Intellisense, guessing this is what they want, and then trying it.